packages/jni-swig-stub/realm.i

```diff
@@ -369,4 +375,93 @@ register_object_notification_cb(realm_object_t *object, jobject callback) {
     );
 }
 
+inline void jni_check_exception(JNIEnv* jenv = realm::jni_util::get_env()) {
+    if (jenv->ExceptionCheck()) {
+        jenv->ExceptionDescribe();
+        throw std::runtime_error("An unexpected Error was thrown from Java.");
+    }
+}
+
+class CustomJVMScheduler{
+    public:
+        CustomJVMScheduler(jobject dispatchScheduler) {
+            JNIEnv *jenv = realm::jni_util::get_env(true);
+            jclass jvm_scheduler_class = jenv->FindClass("io/realm/internal/interop/JVMScheduler");
```

Comment:
If you do this via `Function2` for instance then you need to store the `CoroutineScope` outside the lambda to avoid re-creating it each time. The `dispatcher` also needs to be stored as a class member so the lambda can capture it...
```Kotlin
val myLambda = {coreNotificationFunctionPointer: Long, callbackUserdataPointer: Long ->
        val scope: CoroutineScope = CoroutineScope(dispatcher) // <---- 
        val function: suspend CoroutineScope.() -> Unit = {
            realmc.invoke_core_notify_callback(coreNotificationFunctionPointer, callbackUserdataPointer)
        }
        scope.launch(
            scope.coroutineContext,
            CoroutineStart.DEFAULT,
            function
        )
        realmc.register_realm_scheduler((config as LongPointerWrapper).ptr, JVMScheduler(dispatcher))
    }
```